octodns_bind/__init__.py

```diff
@@ -66,6 +66,11 @@ def populate(self, zone, target=False, lenient=False):
             'populate:   found %s records', len(zone.records) - before
         )
 
+        if target:
+            # When acting as a target we ignore any existing records so that we
+            # create a completely new copy
+            return False
+
```

Comment:
It would be preferable to modify `zone_exists` in `ZoneFileProvider` to return `False`.  As is this change would impact `AxfrPopulate` too which requires that the zone exists and thus its `zone_exists` is hard-coded to `True`. 

 https://github.com/octodns/octodns-bind/blob/6b55749ae2a9b675bec6ee551b6b5e93fcc84489/octodns_bind/__init__.py#L207-L209

Reply:
Thanks for pointing that out!
I implemented a alsmost stub populate() in ZoneFileProvider. zone_exists does not work, since in that method we do not know if we are target or source.

Reply:
Should be fine to pass `target` into the `zone_populate`.

Reply:
Do you think it would be better than my latest commit?

Reply:
Yeah. I think having `zone_exists` is preferable to overriding populate. Having both is pretty confusing since the return value of `zone_exists` is returned. 

It should be clean to add your original logic

```python
        if target:
            # When acting as a target we ignore any existing records so that we
            # create a completely new copy
            return False
```

to https://github.com/octodns/octodns-bind/blob/6477851f9bf8eb1800b9de62d136ac69d0fab9cf/octodns_bind/__init__.py#L212.